app/controllers/price_policies_controller.rb

```diff
@@ -6,193 +6,176 @@ class PricePoliciesController < ApplicationController
+        .for_date(@start_date)
+        .first
+    )
+  end
 
-    raise ActiveRecord::RecordNotFound unless original_price_policies.all?{ |pp| pp.editable? } || original_price_policies.empty?
-    groups_with_policy = Hash[original_price_policies.map {|pp| [pp.price_group, pp] }]
-    current_facility.price_groups.each do |pg|
-      @price_policies << (groups_with_policy[pg] || model_class.new({:price_group_id => pg.id, :product_id => @product.id, :can_purchase => false }))
-    end
+  def init_product
+    @product = current_facility.method(product_var.pluralize)
+      .call
+      .find_by_url_name!(params["#{product_var}_id".to_sym])
+    instance_variable_set("@#{product_var}", @product)
   end
 
-  def update_policies_from_params
-    @price_policies.each do |price_policy|
-      pp_param = params["price_policy_#{price_policy.price_group.id}"] || {:can_purchase => false}
-      pp_param.merge!(
-        start_date: parse_usa_date(params[:start_date]).beginning_of_day,
-        expire_date: parse_usa_date(@expire_date).end_of_day
-      )
-      pp_param.merge! charge_for: params[:charge_for] if params[:charge_for].present?
-      price_policy.attributes = pp_param
-    end
+  def last_price_policy_for_price_group(price_group)
+    @product
+      .price_policies
+      .where(price_group_id: price_group.id)
+      .order(:expire_date)
+      .last
   end
 
+  def make_all_price_policies_purchaseable
+    @price_policies.each { |price_policy| price_policy.can_purchase = true }
+  end
 
   def model_name
-    self.class.name.gsub('Controller', '').singularize
+    self.class.name.gsub(/Controller\z/, "").singularize
   end
-  def model_class
-    model_name.constantize
+
+  def new_start_date
+    # If there are active policies, start tomorrow. If none, start today
+    Date.today + (active_policies? ? 1 : 0)
   end
 
+  def product_var
+    @product_var ||= model_name.gsub(/PricePolicy\z/, "").downcase
+  end
 
-  #
-  # Override CanCan's find -- it won't properly search by zoned date
-  def init_price_policy
-    product_var="@#{model_name.gsub('PricePolicy', '').downcase}"
+  def remove_active_policy_warning
+    flash[:error] =
+      I18n.t("controllers.price_policies.errors.remove_active_policy")
+    redirect_to facility_product_price_policies_path
+  end
 
-    instance_variable_set(
-      "@#{model_name.underscore}",
-      instance_variable_get(product_var).price_policies.for_date(start_date_from_params).first
-    )
+  def set_expire_date_from_params
+    @expire_date = params[:expire_date]
   end
 
+  def set_max_expire_date
+    @max_expire_date = PricePolicy.generate_expire_date(@start_date)
+  end
 
   def start_date_from_params
-    start_date=params[:id] || params[:start_date]
-    return unless start_date
-    format=start_date.include?('/') ? "%m/%d/%Y" : "%Y-%m-%d"
-    Date.strptime(start_date, format)
+    start_date = params[:id] || params[:start_date] || return
+    format = start_date.include?("/") ? :usa : :ymd
```

Comment:
Good point: I don't think this is generally possible given the UI will not even let me type a `-` character into that field. I'll take it out.